my-app/src/App.tsx

```diff
@@ -1,21 +1,52 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
-import { MapContainer, TileLayer } from 'react-leaflet'
+import { useEffect, useState } from 'react';
+import { MapContainer, TileLayer, Polyline } from 'react-leaflet';
 import 'leaflet/dist/leaflet.css';
+import './App.css';
 
-function App() {
+import { LocationMarker } from './components/LocationMarker/LocationMarker';
+
+type LatLngTuple = [number, number];
+
+const start: LatLngTuple = [34.7818, 32.0853]; // Tel Aviv
+const end: LatLngTuple = [35.2137, 31.7683];   // Jerusalem
+
+export const  App = () => {
+  const [route, setRoute] = useState<LatLngTuple[] | null>(null);
+
+  useEffect(() => {
+    let fetched = false;
+
+    const fetchRoute = async () => {
+      try {
+        const url = `http://localhost:8000/route?start=${start.join(',')}&end=${end.join(',')}`;
+        const res = await fetch(url);
+        if (!res.ok) throw new Error(`status ${res.status}`);
+        const data = await res.json();
+        if (!fetched) setRoute(data.route);
+      } catch (err) {
+        console.error('Failed to fetch route:', err);
+      }
+    };
+
+    fetchRoute();
+    return () => {
+      fetched = true;
```

Comment:
This logic is very weird. You run an async function without awaiting it's result and count on the fact that since the useEffect will end first the let variable fetched will be updated before the request is finished. Change this so it's more robust